tinygrad/tensor.py

```diff
@@ -2280,16 +2282,13 @@ def _broadcasted(self, y:Union[Tensor, ConstType], reverse:bool=False, match_dty
       else: y_dtype = dtypes.from_py(y)
       if isinstance(y, Node): y = Tensor.from_node(y, device=self.device)
       else: y = Tensor(dtypes.as_const(y, y_dtype), self.device, y_dtype, requires_grad=False)
-
-    if match_dtype:
+    if match_dtype and not x.dtype == y.dtype:
```

Comment:
is `not x.dtype == y.dtype` and `not x.shape == y.shape` faster?

Reply:
No, should be the same as `x.dtype != y.dtype`.